chain/network-primitives/src/network_protocol/mod.rs

```diff
@@ -454,3 +454,24 @@ impl StateResponseInfo {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::net::IpAddr;
+    use std::net::SocketAddr;
+    use std::net::{Ipv4Addr, Ipv6Addr};
+
+    #[test]
+    fn test_from_str() {
+        use crate::types::PeerInfo;
+        use std::str::FromStr;
+        let peer_test = PeerInfo::from_str(
+            "ed25519:C6HLP37VJN1Wj2irxxZPsVsSya92Rnx12tqK3us5erKV@localhost:1337",
```

Comment:
Technically, what "localhost" resolves to depends on /etc/hosts, so there is no strong guarantee that it will be 127.0.0.1. Please add a TODO to this test that it might require an improvement (probably by mocking the DNS resolution).